### PR TITLE
Add Filter/Hook

### PR DESCRIPTION
**Pull request by dshanske**
Adds a filter to change the arguments passed to wp_insert_post and an action hook to perform actions after the post is updated using the micropub data.

### CODE REVIEW

**Comment by aaronpk** [comment]
Isn't there already a wordpress hook that runs after a post is updated? I would assume it would also run after this plugin updates the post.

**Comment by dshanske** [comment]
But it wouldn't pass the micropub data to it.

**Comment by snarfed** [comment]
right, that's the useful addition.

**Comment by aaronpk** [comment]
cool! (I don't know very much about wordpress :smile:)

File: micropub.php [hunk 1]
```diff
@@ -80,6 +79,7 @@ public static function parse_query($wp) {
     if (!isset($q['url']) || $q['action'] == 'create') {
       $args['post_status'] = 'publish';
       $result = Micropub::check_error(wp_insert_post($args));
+      $args['ID'] = $result;
```
Comment (snarfed) [inline micropub.php]: this can be removed

File: micropub.php [hunk 2]
```diff
@@ -56,10 +56,9 @@ public static function parse_query($wp) {
     if (!array_key_exists('micropub', $wp->query_vars)) {
       return;
     }
-
     $input = file_get_contents('php://input');
     parse_str($input, $q);
-
+
```
Comment (snarfed) [inline micropub.php]: nit: whitespace

File: micropub.php [hunk 3]
```diff
@@ -95,6 +95,7 @@ public static function parse_query($wp) {
         status_header(204);
       } elseif ($q['action'] == 'delete') {
         Micropub::check_error(wp_trash_post($args['ID']));
+	$result = $arg['ID']);
```
Comment (snarfed) [inline micropub.php]: this can be removed

File: micropub.php [hunk 4]
```diff
@@ -111,8 +112,8 @@ public static function parse_query($wp) {
         exit;
       }
     }
-
-    do_action('micropub_request', $q);
+
```
Comment (snarfed) [inline micropub.php]: nit: whitespace

File: micropub.php [hunk 5]
```diff
@@ -187,7 +188,7 @@ private static function map_params($q) {
       $args['post_date'] = iso8601_to_datetime($q['published']);
     }
 
-    return $args;
+    return apply_filter('before_micropub', $args, $q);
```
Comment (snarfed) [inline micropub.php]: mind moving this into `parse_query`, after the `map_params` call? it's subjective, but i'd prefer it there.

**Pull request merged**